editor/src/components/editor/store/dispatch.tsx

```diff
@@ -742,6 +750,35 @@ function editorDispatchInner(
       }
     }
 
+    const actionNames = simpleStringifyActions(dispatchedActions)
+
+    // Check for duplicate UIDs that have originated from actions being applied.
+    const uniqueIDsResult = getAllUniqueUids(result.unpatchedEditor.projectContents)
+    if (uniqueIDsResult.duplicateIDs.length > 0) {
+      const errorMessage = `Running ${actionNames} resulted in duplicate UIDs ${JSON.stringify(
+        uniqueIDsResult.duplicateIDs,
+      )}.`
+      //if (IS_TEST_ENVIRONMENT) {
+      // In tests blow out with an exception so that the error is correctly attributed.
+      //  throw new Error(errorMessage)
+      //} else {
+      // When running in the browser log the error and tell the user to restart the editor.
```

Comment:
I think we can delete this comment

Reply:
I was thinking that it would still be preferable to have this behaviour, albeit only if we can fix the browser tests to not fail silently. As it requires us to scan the big pile of logging when running the browser tests to identify any cases that now fail.

But I'm happy for the wisdom of the crowd to determine if this should just be obliterated until such a time as our tests do capture these exceptions.